values/values.go

```diff
@@ -309,6 +309,24 @@ func CheckKind(got, exp semantic.Nature) {
 	}
 }
 
+type unexpectedKind struct {
+	got, want semantic.Nature
+}
+
+func (e unexpectedKind) Error() string {
+	var err *errors.Error
+	e.As(&err)
```

Comment:
Why are you using `As` here instead of just returning the message? I figure you must have a reason for this but I'm not sure what it is.

Reply:
I didn't want to duplicate the code.